cmd/juju/commands/upgradejuju.go

```diff
@@ -276,7 +276,7 @@ func (c *upgradeJujuCommand) Run(ctx *cmd.Context) (err error) {
 		// environment version (can't guarantee API compatibility).
 		return errors.Errorf("cannot upgrade a %s model with a %s client",
 			agentVersion, jujuversion.Current)
-	case c.Version != version.Zero && c.Version.Major < agentVersion.Major:
+	case c.Version != version.Zero && compareNoBuild(agentVersion, c.Version) == 1:
```

Comment:
So I think the issue is that the check here is just Major and it needed to be at least Major && Minor. But it *shouldn't* be Major.Minor.Patch because I believe we *do* let you roll back patch versions.